Dockerfile

```diff
@@ -1,37 +1,80 @@
-FROM kuzzleio/base
+################################################################################
+# Plugin development image
+################################################################################
+FROM debian:stretch-slim as plugin-dev
 
 LABEL io.kuzzle.vendor="Kuzzle <support@kuzzle.io>"
-LABEL description="Power your web, mobile & iot apps with the Kuzzle backend"
+LABEL description="Develop new plugin or protocol for Kuzzle with ease"
 
-COPY ./ /var/app/
-COPY ./docker-compose/scripts/run.sh /usr/local/bin/kuzzle
+ENV NODE_VERSION=8.9.0
+ENV PATH=/opt/node-v$NODE_VERSION-linux-x64/bin:$PATH
+
+ADD https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz /tmp/
+
+RUN  set -x \
+  \
+  && apt-get update && apt-get install -y \
+       bash-completion \
+       build-essential \
+       curl \
+       g++ \
+       gdb \
+       git \
+       python \
+       libfontconfig \
+       wget \
+  && tar xf /tmp/node-v$NODE_VERSION-linux-x64.tar.gz -C /opt/ \
+  && rm /tmp/node-v$NODE_VERSION-linux-x64.tar.gz \
+  && mkdir -p /var/app \
+  && npm install -g npm \
+  && npm set progress=false \
+  && npm install -g \
+    pm2 \
+  && echo "" > /opt/node-v$NODE_VERSION-linux-x64/lib/node_modules/pm2/lib/keymetrics \
+  && rm -rf /var/lib/apt/lists/* \
+  && echo "alias ll=\"ls -lahF --color\"" >> ~/.bashrc
+
+ARG kuzzle_tag
+
+RUN  git clone --depth 1 --branch $kuzzle_tag https://github.com/kuzzleio/kuzzle /var/app \
+  && git clone --depth 1 https://github.com/kuzzleio/kuzzle-plugin-auth-passport-local /var/app/plugins/available/kuzzle-plugin-auth-passport-local \
```

Comment:
what if master branch of kuzzle-plugin-auth-passport-local is not compatible with `$kuzzle_tag` version of kuzzle ?

Reply:
Well, we have a problem then I think.  
I didn't figure out a simple solution.. I just assume these plugin will be compatible with Kuzzle when the image will be built.  
When the `2.x` come out we will have to think to a compatibility matrix or something

Reply:
@ballinette > the way I see it, official plugins must be compatible with all available versions of Kuzzle at the moment of the plugin's release.
This means that we'll have to have fallbacks depending on the target Kuzzle version.

I think this is way easier that way, instead of maintaining multiple plugin major versions + compatibility matrixes everywhere.

Reply:
But there is no reason to clone these plugins separately from Kuzzle core as they are already included as submodule (so `git clone --recursive --depth 1 --branch $kuzzle_tag https://github.com/kuzzleio/kuzzle /var/app` should be sufficient)

... and for the next major release, my advice is that:
* plugin passport local should not be forced anymore (we can want to not authenticate with login/password, but with other strategies)
* plugin logger should not be a plugin anymore and integrated within the core (winston is configurable enough to set up the logger as we want, without need to create another plugin)

Reply:
We talked about the submodules and we weren't sure we'd keep them, I think.
At least with the git clone we decide exactly what we want in the image